In [52]:
import pandas as pd
import numpy as np

# Set the random seed for reproducibility
np.random.seed(42)

# Generate synthetic data for a finance investment sector
data = {
    'Investor_ID': np.arange(1, 101),
    'Investment_Type': np.random.choice(['Stocks', 'Bonds', 'Real Estate', 'Mutual Funds', 'Cryptocurrency'], size=100),
    'Investment_Amount': np.round(np.random.uniform(1000, 50000, size=100), 2),
    'ROI_Percentage': np.round(np.random.uniform(-10, 25, size=100), 2),  # Return on Investment (ROI) percentage
    'Investment_Duration': np.random.randint(1, 10, size=100),  # Duration in years
    'Risk_Level': np.random.choice(['Low', 'Medium', 'High'], size=100)
}

# Create a pandas DataFrame
df = pd.DataFrame(data)

# Show the first few rows of the dataset
df.head()


Investor_ID Investment_Type  Investment_Amount  ROI_Percentage  \
0            1    Mutual Funds           45833.02           13.65   
1            2  Cryptocurrency           42651.89           15.73   
2            3     Real Estate           23023.08           -2.68   
3            4  Cryptocurrency            5675.10            8.95   
4            5  Cryptocurrency           19170.09           14.35   

   Investment_Duration Risk_Level  
0                    3        Low  
1                    7       High  
2                    2     Medium  
3                    2        Low  
4                    7     Medium


### Dataset Columns:
- **Investor_ID**: Unique ID for each investor.
- **Investment_Type**: Type of investment (e.g., Stocks, Bonds, Real Estate, Mutual Funds, Cryptocurrency).
- **Investment_Amount**: The amount invested (in USD).
- **ROI_Percentage**: The return on investment (ROI) as a percentage.
- **Investment_Duration**: Duration of the investment (in years).
- **Risk_Level**: The associated risk level of the investment (Low, Medium, High).

### Sample Data:

| Investor_ID | Investment_Type | Investment_Amount | ROI_Percentage | Investment_Duration | Risk_Level |
|-------------|-----------------|-------------------|----------------|---------------------|------------|
| 1           | Mutual Funds     | 45833.02          | 13.65          | 3                   | Low        |
| 2           | Cryptocurrency   | 42651.89          | 15.73          | 7                   | High       |
| 3           | Real Estate      | 23023.08          | -2.68          | 2                   | Medium     |
| 4           | Cryptocurrency   | 5675.10           | 8.95           | 2                   | Low        |
| 5           | Cryptocurrency   | 19170.09          | 14.35          | 7                   | Medium     |

### 10 Query Questions:

#### Easy:
1. **Total Investment per Type**: What is the total investment amount for each type of investment?
2. **Average ROI by Risk Level**: What is the average ROI percentage for each risk level (Low, Medium, High)?
3. **Investors with High ROI**: Find all investors who have an ROI percentage greater than 10%.

#### Medium:
4. **Top 5 Investors by Investment Amount**: Identify the top 5 investors who have invested the highest amounts.
5. **Average Investment by Investment Duration**: What is the average investment amount for each duration of investment (1-9 years)?
6. **Risk Level Distribution**: How many investors are there in each risk level (Low, Medium, High)?
7. **Investments in Real Estate**: What is the total investment and average ROI for investors who invested in "Real Estate"?

#### Advanced:
8. **Correlation Between ROI and Investment Amount**: Is there any correlation between the amount invested and the ROI percentage?
9. **Top Investors by Risk Level**: For each risk level, find the investor with the highest investment amount.
10. **Return on Investment over Duration**: For each investment duration, calculate the average ROI percentage and analyze trends over time.

### The Story of “InvestSmart”: Unveiling Investment Insights for Optimal Growth

#### Background:
In the competitive world of finance, “InvestSmart,” a leading investment advisory firm, aimed to better understand their clientele of 100 investors. The company sought to enhance its offerings by providing tailored investment strategies. However, they faced a significant challenge: a lack of clarity regarding investment behaviors, ROI performance, and risk management across various investment types. With an expansive dataset at their disposal, they embarked on a mission to uncover actionable insights.

#### Problem Statement:
InvestSmart realized that despite their extensive client base, they struggled to derive meaningful conclusions from the data. They needed to answer crucial questions, such as:

1. What are the investment preferences of their clients?
2. How does ROI vary by risk level?
3. Which investors have the potential for higher returns, and how can they tailor strategies accordingly?

The data consisted of key columns like **Investor_ID**, **Investment_Type**, **Investment_Amount**, **ROI_Percentage**, **Investment_Duration**, and **Risk_Level**. With these, they set out to explore various dimensions of their investments.

#### Conclusion:
Through rigorous analysis and strategic querying, InvestSmart transformed a dataset into a powerful decision-making tool. The insights gained not only enhanced their understanding of client behaviors but also refined their investment strategies. By addressing their initial challenges, InvestSmart positioned itself as a data-driven advisory firm, ready to navigate the complexities of investment management. This proactive approach not only strengthened client relationships but also bolstered their reputation in the industry.

#### The Impact:
InvestSmart’s investment strategies became more aligned with their clients' goals, leading to improved satisfaction and retention rates. By embracing data analytics, they unlocked the potential for smarter investment decisions, proving that informed strategies lead to lasting success in the financial world.


In [53]:
# This is the funtion which convert the output in dataframe just to maintain its beauty
def pretty_view(df,option=None):
    if option == 'Yes':
        return pd.DataFrame(df).reset_index()
    if option == 'No':
        return pd.DataFrame(df)
    
# This is th function for a alias purposes
def change_name(df,new_names):
    return df.rename(columns=new_names)
    


In [54]:
# 1. Total Investment per Type: What is the total investment amount for each type of investment?

invest_cate =df.groupby('Investment_Type')['Investment_Amount'].sum()
invest_cate.columns=['Investment_Type','Investment_Amount']
pretty_view(invest_cate,option='Yes')


Investment_Type  Investment_Amount
0           Bonds          498042.25
1  Cryptocurrency          551327.26
2    Mutual Funds          587864.32
3     Real Estate          412564.72
4          Stocks          431412.74

### Analysis -1
 Mutual Funds are the top investment type where around **$587864** has been invested. Following that Real state has surprisingly became the least investment type with the accumulate capital of **$412564**, whereas crypto,stocks and bonds have the following data  **$551327**, **$431412**, **$498042** respectively.

In [55]:
# 2. Average ROI by Risk Level: What is the average ROI percentage for each risk level (Low, Medium, High)?

roi_avg=df.groupby('Risk_Level')['ROI_Percentage'].mean().round(0)
pretty_view(roi_avg,option='Yes')


Risk_Level  ROI_Percentage
0       High             9.0
1        Low             8.0
2     Medium             8.0

### Analysis -2
Approximately **9%** of the ROI is associated with high risk, while **7%** falls under low risk. In contrast, the medium risk level accounts for **8%** of the ROI.

In [56]:
# 3. Investors with High ROI: Find all investors who have an ROI percentage greater than 10%.
high_roi=df[df['ROI_Percentage']>10]
high_roi=high_roi[['Investor_ID','ROI_Percentage']]
pretty_view(high_roi.head(),option='No')

Investor_ID  ROI_Percentage
0             1           13.65
1             2           15.73
4             5           14.35
7             8           24.38
10           11           24.87

### Analysis -3
Among 100 of investors only **43** investors have ROI of more than **10%**

In [57]:
# 4. Top 5 Investors by Investment Amount: Identify the top 5 investors who have invested the highest amounts.

rich_investors=df.sort_values(['Investment_Amount'],ascending=False)
rich_investors=rich_investors[['Investor_ID','Investment_Amount']]
pretty_view(rich_investors.head(),option='No')


Investor_ID  Investment_Amount
65           66           49889.28
82           83           49187.73
67           68           48854.13
11           12           48613.89
29           30           47199.82

### Analysis -4
The top five investors have invested amounts exceeding 46,000, with the highest investment being 49,889 and the lowest among them being 47,199.

In [58]:
# 5. Average Investment by Investment Duration: What is the average investment amount for each duration of investment (1-9 years)?

invest_duration=df.groupby('Investment_Duration')['Investment_Amount'].mean().round(2)
pretty_view(invest_duration,option='No')

Investment_Amount
Investment_Duration                   
1                             30917.29
2                             20201.78
3                             31077.20
4                             24313.25
5                             18795.66
6                             28759.65
7                             19774.04
8                             25204.52
9                             21191.13

### Analysis -5
The longest investment duration was **9 years**, while the shortest was **1 year**. The average investment amounts for each duration are summarized in the report above.

In [59]:
# 6. Risk Level Distribution: How many investors are there in each risk level (Low, Medium, High)?
investor_risk_lvl=df.groupby('Risk_Level')['Investor_ID'].count().reset_index()
before_final=pretty_view(investor_risk_lvl.head(),option='No')
# df.head()

#Changing the columns name
new_names={'Investor_ID':'Total_investors'}
renamed=change_name(before_final,new_names) #special fx made for changing name
renamed

Risk_Level  Total_investors
0       High               30
1        Low               45
2     Medium               25

### Analysis -6
This indicates that most investors favor low-risk opportunities, while a smaller portion of the group is divided between medium and high-risk investments.

In [60]:
# 7. Investments in Real Estate: What is the total investment and average ROI for investors who invested in "Real Estate"?
# Extarcting investment type
investment_area=df[df['Investment_Type']=='Real Estate']

#Extracting total investment
total_money= investment_area['Investment_Amount'].sum().round(2)

#Extarcting average roi
avg_roi=investment_area['ROI_Percentage'].mean().round(2)


#Final Result
print(f'The total Investment Amount in Real estate Industry is ${total_money}')
print(f'The Average ROI for Real estate Industry is {avg_roi}%')

The total Investment Amount in Real estate Industry is $412564.72
The Average ROI for Real estate Industry is 6.11%


### Analysis -7
The total investment in the Real Estate industry amounts to $412,564.72, indicating significant interest and capital allocation in this sector. The average return on investment (ROI) for these investments stands at 6.11%, reflecting a moderate performance level for investors in the Real Estate industry.

In [61]:
# 8. Correlation Between ROI and Investment Amount: Is there any correlation between the amount invested and the ROI percentage?

correlation = df['Investment_Amount'].corr(df['ROI_Percentage'])
print(f'Correlation between Investment Amount and ROI Percentage: {correlation.round(2)}')


Correlation between Investment Amount and ROI Percentage: 0.04


###  Analysis: -8

The **correlation coefficient**  calculated between the **Investment Amount** and **ROI Percentage** is **0.04**, which is very close to **0**. This means:

- There is **almost no correlation** between the investment amount and the return on investment (ROI) percentage.
- In other words, as the amount invested changes, it does not strongly affect the ROI percentage, either positively or negatively.


In [62]:
# Top Investors by Risk Level: For each risk level, find the investor with the highest investment amount.

#top investment amount
investment_details=df.loc[df.groupby('Risk_Level')['Investment_Amount'].idxmax()]

#investors id details
investors= investment_details[['Investor_ID','Investment_Type','Investment_Amount','Risk_Level']]
investors

Investor_ID Investment_Type  Investment_Amount Risk_Level
65           66    Mutual Funds           49889.28       High
11           12     Real Estate           48613.89        Low
82           83  Cryptocurrency           49187.73     Medium

### Analysis -9

- **High Risk**: Investor **66** has the highest investment in **Mutual Funds** with **$49,889.28**.
- **Medium Risk**: Investor **83** leads with **$49,187.73** in **Cryptocurrency**.
- **Low Risk**: Investor **12** tops with **$48,613.89** in **Real Estate**.

Each investor represents the highest investment in their respective risk level.

In [63]:
# 10. **Return on Investment over Duration**: For each investment duration, calculate the average ROI percentage and analyze trends over time.
# Assuming df is your original DataFrame
invest_dur = df.groupby('Investment_Duration')['ROI_Percentage'].mean().round(0)

# Resetting index and adding 'Years' to the Investment Duration
invest_dur = invest_dur.reset_index()
invest_dur['Investment_Duration'] = invest_dur['Investment_Duration'].astype(str) + ' Years'

# Optionally, if you want to rename the mean ROI column for clarity
invest_dur.columns = ['Investment_Duration', 'Average_ROI in %']

# Display the results
pretty_view(invest_dur, option='No')


Investment_Duration  Average_ROI in %
0             1 Years               7.0
1             2 Years              10.0
2             3 Years               4.0
3             4 Years              12.0
4             5 Years               9.0
5             6 Years              10.0
6             7 Years               9.0
7             8 Years               3.0
8             9 Years              15.0

### Analysis -10
- **Highest ROI**: **9 Years** at **15.0%**.
- **Lowest ROI**: **8 Years** at **3.0%**.
- **Trend**: Longer durations generally yield higher returns, with fluctuations observed.

